rust/ethers-prometheus/src/json_rpc_client.rs

```diff
@@ -74,7 +74,16 @@ pub struct PrometheusJsonRpcClientConfig {
     pub chain: Option<ChainInfo>,
 }
 
-impl PrometheusJsonRpcClientConfig {
+/// Helper functions for displaying node and chain information
+pub trait PrometheusJsonRpcClientConfigExt {
```

Comment:
What is the benefit to this being an ext trait vs just added to the `PrometheusJsonRpcClientConfig` impl? We should have full control over the original so I think it makes more sense in this case to update it.

Reply:
I wasn't able to test the fallback provider logic previously with the concrete `PrometheusJsonRpcClientConfig` type. Now that it uses a generic bound by `PrometheusJsonRpcClientConfigExt`, I could just implement my own `ProviderMock` to test the new fallback provider logic (see the unit tests at the end of `fallback.rs`).